forem/forem
# <MultiSelectAutocomplete /> component only allows for fetched suggestions now

## What type of PR is this? (check all applicable)


## Description

Now the `<MultiSelectAutocomplete />` only has the responsibility of allowing existing fetched suggestions to be added as items to the list. Previously, it allowed for new items to be added that were specific to the tags field in the article editor.

From @aitchiss in https://github.com/forem/forem/pull/16604#discussion_r831935722

> Just an idea, but I think perhaps the `MultiSelectAutocomplete` shouldn't be responsible for this.
> 
> So currently:
> 
> 1. We pass in `fetchSuggestions` which returns only existing matching tags from the search call
> 2. In `MultiSelectAutocomplete` if the results are empty, we add the current search term to the suggestions
> 
> Instead we could:
> 
> * Not do point 2 above
> * In `TagsField` which we use for the editor, we could make sure the `fetchSuggestions` prop we pass in adds the current search term as a suggestion if nothing was found in the search call

## Related Tickets & Documents



- Related Issue #16604
- Closes #

## QA Instructions, Screenshots, Recordings

1. Create a new blog post by navigating to http://localhost:3000/new
2. In the tag field, add the existing tag `java`.
3. A dropdown appears with two options.
4. Select `java` from the dropdown or press `,` or the spacebar to insert it into the tags list.
5. Enter a tag that does not exist, e.g. `asdf`
6. 3. A dropdown appears which includes `asdf` in the list.
7. Select `asdf` from the dropdown or press `,` or the spacebar to insert it into the tags list.

### UI accessibility concerns?

Accessibility of the component should be the same as it currently is.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Jimmy Fallon saying "Better, faster, stronger](https://media.giphy.com/media/QuP2I5u0tcY7v6LCpp/giphy.gif)
